apps/performance/ngfor-optimize/src/app/app.component.ts

```diff
@@ -50,7 +50,7 @@ export class AppComponent implements OnInit {
   }
 
   handleKey(event: any) {
-    if (event.keyCode === 13) {
+    if (event.keyCode === 13 && this.label.length) {
```

Comment:
Instead of checking the keyCode inside the method, you can utilize angulars integrated (keydown.enter) directive. Just replace `(keydown)="handleKey($event)"` with `(keydown.enter)="handleKey($event)"`. [Check this out](https://www.digitalocean.com/community/tutorials/angular-binding-keyup-keydown-events).